crates/walrus-service/src/node/blob_event_processor.rs

```diff
@@ -424,6 +424,13 @@ impl BlobEventProcessor {
             // certified event processing, as certified events take longer and can block following
             // registered events.
             let _scope = monitored_scope::monitored_scope("ProcessEvent::BlobEvent::Registered");
+            if let Err(error) = self.node.flush_pending_caches(&event.blob_id).await {
```

Comment:
Important: with the locking, this for sure will affect event processing throughput. This should be done in the worker above.